### PR TITLE
Automate Releases based on Tagging

### LINKED ISSUE #1406: Add auto github release workflow
### Is your feature request related to a problem?
As part of OpenSearch releases, `opensearch-build` pushes a tag for the release commit.
But publishing a github release based on the tag is manual.

### What solution would you like?
We've added this workflow[1] in OpenSearch which publishes github releases automatically when a tag is pushed.

Add this workflow to all repos releasing software in the project.
See example: https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/auto-release.yml
Example Release: https://github.com/opensearch-project/OpenSearch/releases/tag/2.5.0

[1] https://github.com/opensearch-project/opensearch-plugins/blob/main/WORKFLOWS.md#auto-github-releases

### PR DESCRIPTION
**Pull request by Swiddis**
### Description
Generating a GitHub Release involves a few steps: selecting the commit, creating a version bump, and creating the release notes file. Per work mentioned in #1406, this has been automated to be performed based on tag selection. This PR:
- Modifies the release label policy in `draft-release-notes-config.yml` to better reflect labels currently in use
	- Includes two new labels, `infrastructure` and `skip-changelog`, that need to be added to the repo
- Implements the auto-release workflow from [OpenSearch/auto-release.yml at main · opensearch-project/OpenSearch](https://github.com/opensearch-project/OpenSearch/blob/1540e00397a19f41d6e92248a3d51d1544255e3f/.github/workflows/auto-release.yml), see `auto-release.yml`
- Adds a policy for enforcing that future PRs are appropriately labeled for more robust release notes, see `enforce-labels.yml`

### Issues Resolved
- Resolves #1406

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/draft-release-notes-config.yml [hunk 1]
```diff
@@ -12,11 +12,17 @@ replacers:
...
+
+exclude-contributors:
+  - 'opensearch-drigger-bot'
+
 # Organizing the tagged PRs into unified categories
 categories:
   - title: 'Breaking Changes'
     labels:
-      - 'Breaking Changes'
+      - 'breaking'
```
Comment (vamsi-amazon) [inline .github/draft-release-notes-config.yml:25]: breaking-changes?

File: .github/draft-release-notes-config.yml [hunk 2]
```diff
@@ -12,11 +12,17 @@ replacers:
   - search: '##'
     replace: '###'
 
+exclude-labels:
+  - 'skip-changelog'
+
+exclude-contributors:
+  - 'opensearch-drigger-bot'
```
Comment (vamsi-amazon) [inline .github/draft-release-notes-config.yml:19]: can you correct the spelling.

**Comment by vamsi-amazon** [comment]
The current state of the draft release is looking like this: https://github.com/opensearch-project/observability/releases/tag/2.1.0.0
Are we in good place for next release. Do we need to any cleanup on the existing draft release.

File: .github/draft-release-notes-config.yml (hunk 1 above)
Comment (Swiddis) [reply .github/draft-release-notes-config.yml:25]: We already have a `breaking` label but no `breaking-changes` one, is there a difference between them that warrants having both?

Comment (Swiddis) [reply .github/draft-release-notes-config.yml:25]: It looks like a `>breaking` label [is used in the OpenSearch repo](https://github.com/opensearch-project/OpenSearch/issues?q=label%3A%3Ebreaking). For consistency maybe we should go with this?

**Review by YANG-DB — DISMISSED** [review]

File: .github/draft-release-notes-config.yml [hunk 3]
```diff
@@ -12,11 +12,17 @@ replacers:
   - search: '##'
     replace: '###'
 
+exclude-labels:
+  - 'skip-changelog'
+
+exclude-contributors:
```
Comment (derek-ho) [inline .github/draft-release-notes-config.yml:18]: do we want to exclude trigger bot? What about backported changes? Especially now that everyone is working off main branch

**Review by derek-ho — APPROVED** [review]
LGTM

**Comment by derek-ho** [comment]
> The current state of the draft release is looking like this: https://github.com/opensearch-project/observability/releases/tag/2.1.0.0 Are we in good place for next release. Do we need to any cleanup on the existing draft release.

This may only be from next release onwards, unless we want to go back and manually tag all PRs going into 2.6

**Review by derek-ho — DISMISSED** [review]
LGTM 🚀  🚀  🚀  🚀  🚀  🚀  🚀  🚀  🚀  🚀  🚀  🚀  🚀  🚀  🚀  🚀  🚀  🚀  🚀  🚀  🚀

**Review by YANG-DB — APPROVED** [review]

**Comment by ps48** [comment]
@Swiddis should we add the `infrastructure` label to this PR?

**Comment by Swiddis** [comment]
> @Swiddis should we add the `infrastructure` label to this PR?

Yeah, [the SQL repo](https://github.com/opensearch-project/sql/labels?q=infrastructure) defines infrastructure as "Changes to infrastructure, testing, CI/CD, pipelines, etc."

File: .github/workflows/auto-release.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+          private_key: ${{ secrets.APP_PRIVATE_KEY }}
+          installation_id: 22958780
+      - name: Get tag
+        id: tag
+        uses: dawidd6/action-get-tag@v1
+      - uses: actions/checkout@v2
+      - uses: ncipollo/release-action@v1
+        with:
+          github_token: ${{ steps.github_app_token.outputs.token }}
+          bodyFile: release-notes/opensearch=observability.release-notes-${{steps.tag.outputs.tag}}.md
```
Comment (derek-ho) [inline .github/workflows/auto-release.yml:29]: Can you fix this = supposed to be dash

**Comment by Swiddis** [comment]
Closing to re-base onto 2.x

**Pull request closed without merging**